torch/nn/modules/loss.py

```diff
@@ -1121,6 +1125,91 @@ def forward(self, input: Tensor, target: Tensor) -> Tensor:
                                ignore_index=self.ignore_index, reduction=self.reduction)
 
 
+class CrossEntropyLossWithSoftLabels(_Loss):
+    r"""This criterion combines :class:`~torch.nn.LogSoftmax` and :class:`~torch.nn.KLDivLoss` in
+    one single class. It is useful when training a classification problem with `C` classes when
+    "soft labels" are required, such as when utilizing label smoothing or blended labels.
+
+    The `input` is expected to contain raw, unnormalized scores for each class and the `target`
+    should match the `input` shape and contain the target distribution over the classes.
+
+    The loss can be described as:
```

Comment:
I got a bit confused as to why L is equal to l_{n, c}. I think what we meant was "The unreduced loss `L` is a 2D tensor such that each element `L_{n, c}` is equal to...

Reply:
https://pytorch.org/docs/stable/generated/torch.nn.KLDivLoss.html seems like a good example to follow

Reply:
> I got a bit confused as to why L is equal to l_{n, c}. I think what we meant was "The unreduced loss L is a 2D tensor such that each element L_{n, c} is equal to...

Good point, the L equal to l_{n, c} notation is arguably just wrong and should be changed- I like the phrasing you chose.

> https://pytorch.org/docs/stable/generated/torch.nn.KLDivLoss.html seems like a good example to follow

I think KLDivLoss is a nice and readable example, but it's hard to follow here - unlike for KLDivLoss, the loss here can't be directly defined element-wise because of the log-softmax.

Reply:
Okay, I decided to follow what `CrossEntropyLoss` does here: they just show the equation for softmax without explicitly defining the indices it is computed over